runbotics/runbotics-scheduler/src/utils/bpmn/findSubprocess.ts

```diff
@@ -0,0 +1,82 @@
+import BpmnModdle, { BaseElement, Definitions } from 'bpmn-moddle';
+import { GeneralAction, IProcess } from 'runbotics-common';
+
+interface CustomExtensionElement {
+    values: {
+        $children: {
+            $body: string;
+            $type: string;
+            type: string;
+            name: string;
+        }[];
+        $type: string;
+    }[];
+}
+
+interface CustomModdleElement
+    extends Pick<BaseElement, 'id' | '$attrs' | '$type'> {
+    implementation: string;
+    extensionElements: CustomExtensionElement;
+}
+
+interface CustomDefinition extends Definitions {
+    elementsById: {
+        [key: BaseElement['id']]: CustomModdleElement;
+    };
+}
+
+const startProcessActionBodyPattern = /\${(.+?)}/;
+
+const isActionStartProcess = (elementId: string, element: unknown) =>
+    elementId.startsWith('Activity') &&
+    element !== null &&
+    element !== undefined &&
+    typeof element === 'object' &&
+    '$attrs' in element &&
+    element.$attrs !== null &&
+    element.$attrs !== undefined &&
+    typeof element.$attrs === 'object' &&
+    'camunda:actionId' in element.$attrs &&
+    typeof element.$attrs['camunda:actionId'] === 'string' &&
+    element.$attrs['camunda:actionId'] === GeneralAction.START_PROCESS;
```

Comment:
whoa
consider breaking this down into smaller parts like

const isObject =  
    elementId.startsWith('Activity') &&
    element !== null &&
    element !== undefined &&
    typeof element === 'object'

const hasAttributes=
'$attrs' in element &&
    element.$attrs !== null &&
    element.$attrs !== undefined &&
    typeof element.$attrs === 'object'

const hasStartEventActionId = 
'camunda:actionId' in element.$attrs &&
    typeof element.$attrs['camunda:actionId'] === 'string' &&
    element.$attrs['camunda:actionId'] === GeneralAction.START_PROCESS;

const isActionStartProcess = 
isObject &&
hasAttributes &&
hasStartEventActionId